### PR TITLE
fix: respect client.needHotEntry and client.needClientEntry

### PR DESCRIPTION
**Pull request by jkzing**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes

### Motivation / Use-Case

In version `4.0.0-beta.2` it still uses old option `injectHot` and `injectClient` to determine whether to inject dev server client, this PR corrects that by respecting new option `client.needHotEntry` and `client.needClientEntry`. 

### Breaking Changes

No

### Additional Info

N/A

### CODE REVIEW

**Review by alexander-akait — DISMISSED** [review]

**Comment by alexander-akait** [comment]
Need fix CI

File: lib/utils/DevServerPlugin.js [hunk 1]
```diff
@@ -158,14 +158,17 @@ class DevServerPlugin {
 
     /** @type {Entry} */
     const additionalEntries = checkInject(
-      options.injectClient,
+      options.client.needClientEntry,
```
Comment (snitin315) [inline lib/utils/DevServerPlugin.js:161]:
```suggestion
      options.client ? options.client.needClientEntry : null,
```

**Review by snitin315 — APPROVED** [review]
Thanks!

**Comment by jkzing** [comment]
Humm, coverage decrease may be caused by newly added line breaks...

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
Thanks

**Pull request merged**